src/tests/ftest/io/parallel_io.py

```diff
@@ -192,3 +241,139 @@ def test_parallelio(self):
+        processes = self.params.get("np", '/run/ior/client_processes/*')
+
+        # Create pools in parallel.
+        for _ in range(self.pool_count):
+            pool_thread = threading.Thread(target=self.create_pool())
+            pool_threads.append(pool_thread)
+        # start container create job
+        for pool_job in pool_threads:
+            pool_job.start()
+        # wait for container create to finish
+        for pool_job in pool_threads:
+            pool_job.join()
+
+        # start dfuse using --svc option only.
+        self.start_dfuse()
+
+        # record free space using statvfs before any data is written.
+        statvfs_info_before = self.statvfs_pool(self.dfuse.mount_dir.value)
+
+        # Create 10 containers for each pool. Container create process cannot
+        # be parallelised as different container create could complete at
+        # different times and get appended in the self.container variable in
+        # unorderly manner, causing problems during the write process.
+        for _, pool in enumerate(self.pool):
+            for _ in range(self.cont_count):
+                self.create_cont(pool)
+
+        # Try to access each dfuse mounted container using ls. Once it is
+        # accessed successfully, go ahead and perform io on that location
+        # using ior. This process of performing io is done in parallel for
+        # all containers using threads.
+        for pool_count, pool in enumerate(self.pool):
+            dfuse_pool_dir = str(self.dfuse.mount_dir.value + "/" + pool.uuid)
+            for counter in range(self.cont_count):
+                cont_num = (pool_count * self.cont_count) + counter
+                dfuse_cont_dir = str(dfuse_pool_dir + "/" +
+                                     self.container[cont_num].uuid)
+                cmd = u"###ls -a {}".format(dfuse_cont_dir)
+                self.execute_cmd(cmd)
+
+                # run ior on all containers
+                test_file = dfuse_cont_dir + "/testfile"
+                self.ior_cmd.test_file.update(test_file)
+                self.ior_cmd.set_daos_params(
+                    self.server_group, pool, self.container[cont_num].uuid)
+                thread = threading.Thread(
+                    target=self.run_ior,
+                    args=(self.get_ior_job_manager_command(), processes, None,
+                          False))
+                threads.append(thread)
+                thread.start()
+
+        # wait for all ior jobs to be finished
+        for job in threads:
+            job.join()
+
+        # Record free space after io
+        statvfs_before_cont_destroy = self.statvfs_pool(
+            self.dfuse.mount_dir.value)
+
+        # Destroy half of the containers from each pool
+        pfinal = 0
+        for count in range(self.cont_count):
+            pinitial = pfinal
+            pfinal = pinitial + (self.cont_count / 2)
```

Comment:
Word of caution. If we upgrade to python 3.x, we need to be careful about how we use divide by "/". It is going to return floating point if the numerator is odd number in this case. To be safe, use "//" which will return the floor value.

Reply:
Done